drsm/README.md

```diff
@@ -6,6 +6,26 @@ SPDX-License-Identifier: Apache-2.0
 -->
 # Distributed Resource Sharing Module (DRSM)
 
+## Block diagram
+
+![DRSM Usage in AMF](/drsm/images/drsm.png)
+
+## Introduction
+It is very common in micro services based architecture to share resources among multiple instances of the microservices.Just to explain the DRSM concepts and design we can take an example of the Access Management Function (AMF) network function. e.g. gNodeB & AMF connect over the N2 interface and NGAP protocol is used for this interface. 3gpp defines the specification of this interface. NGAP uses SCTP transport protocol for communication between gNodeB & AMF. Multiple User Equipment (UE) signaling traffic is sent over the N2 interface. Users are identified by NGAPIDs. Each subscriber is identified by the NGAAP ID of AMF & NGAAP ID of gNodeB.
+So it's very important for AMF, gNodeB to assign a unique NGAPP ID to UE.  When single AMF instance is running in the system then there is no issue and assigned IDs are unique. But when multiple instances of AMF are running in the system then the DRSM module helps the AMF instance for unique ID assignment. 
```

Comment:
Please remove all trailing whitespaces from your changes. Otherwise, it is not possible to do `git apply`, as you can see below
```
$ git apply pr.patch
pr.patch:15: trailing whitespace.
So it's very important for AMF, gNodeB to assign a unique NGAPP ID to UE.  When single AMF instance is running in the system then there is no issue and assigned IDs are unique. But when multiple instances of AMF are running in the system then the DRSM module helps the AMF instance for unique ID assignment.
pr.patch:17: trailing whitespace.
## Module
pr.patch:23: trailing whitespace.
## Load Balancer
pr.patch:27: trailing whitespace.
If any AMF instances crashes then the chunks owned by the crashed AMF instances are claimed by other running instances and this functionality is part of DRSM. This way load is distributed across remaining AMF instances after the redundancy event. Load balancer learns about change in owner and future messages are sent to updated owner.  In case a message is not received at the right AMF instance then for brief period of time redirect functionality gets exercised.
error: cannot apply binary patch to 'drsm/images/drsm.png' without full index line
error: drsm/images/drsm.png: patch does not apply
```

Reply:
please check.

Reply:
I still get the same issue
```
$ git apply pr.patch
pr.patch:16: trailing whitespace.
## Module
pr.patch:22: trailing whitespace.
## Load Balancer
pr.patch:26: trailing whitespace.
If any AMF instances crashes then the chunks owned by the crashed AMF instances are claimed by other running instances and this functionality is part of DRSM. This way load is distributed across remaining AMF instances after the redundancy event. Load balancer learns about change in owner and future messages are sent to updated owner.  In case a message is not received at the right AMF instance then for brief period of time redirect functionality gets exercised.
error: cannot apply binary patch to 'drsm/images/drsm.png' without full index line
error: drsm/images/drsm.png: patch does not apply
```

Reply:
Here is the git diff that resolves the issue:
```diff
diff --git a/drsm/README.md b/drsm/README.md
index e0f153e..76f45ca 100644
--- a/drsm/README.md
+++ b/drsm/README.md
@@ -13,17 +13,17 @@ SPDX-License-Identifier: Apache-2.0
 ## Introduction
 It is very common in micro services based architecture to share resources among multiple instances of the microservices.Just to explain the DRSM concepts and design we can take an example of the Access Management Function (AMF) network function. e.g. gNodeB & AMF connect over the N2 interface and NGAP protocol is used for this interface. 3gpp defines the specification of this interface. NGAP uses SCTP transport protocol for communication between gNodeB & AMF. Multiple User Equipment (UE) signaling traffic is sent over the N2 interface. Users are identified by NGAPIDs. Each subscriber is identified by the NGAAP ID of AMF & NGAAP ID of gNodeB. So it's very important for AMF, gNodeB to assign a unique NGAPP ID to UE.  When single AMF instance is running in the system then there is no issue and assigned IDs are unique. But when multiple instances of AMF are running in the system then the DRSM module helps the AMF instance for unique ID assignment.

-## Module
+## Module
 DRSM is a go module and can be used by go program to start using the resource allocation functionality. In 5G core it is used by SCTP load balancer & AMF.

 ## Initialization
 During DRSM module init, AMF provides all required information to DRSM and reserves a few unique chunks. This is done through mongodb operation internally. The DRSM module provides APIs to assign IDs from these chunks. When Chunk is fully utilized then AMF creates a new unique chunk and starts assigning the Ids from the newly created chunk.

-## Load Balancer
+## Load Balancer
 SCTP load balancer keeps track of all the created chunks and their corresponding owner. When UE messages are received at SCTP load balancer then it reads the NGAPP id from the messages and finds the corresponding AMF instances using DRSM local lookup APIs.  This way the same AMF continues to get the same subscriber traffic and we achieve subscriber pinning to specific instances.

 ## During redundancy event
-If any AMF instances crashes then the chunks owned by the crashed AMF instances are claimed by other running instances and this functionality is part of DRSM. This way load is distributed across remaining AMF instances after the redundancy event. Load balancer learns about change in owner and future messages are sent to updated owner.  In case a message is not received at the right AMF instance then for brief period of time redirect functionality gets exercised.
+If any AMF instances crashes then the chunks owned by the crashed AMF instances are claimed by other running instances and this functionality is part of DRSM. This way load is distributed across remaining AMF instances after the redundancy event. Load balancer learns about change in owner and future messages are sent to updated owner.  In case a message is not received at the right AMF instance then for brief period of time redirect functionality gets exercised.

 ## Resources can be
     - integer numbers (TEID, SEID, NGAPIDs,TMSI,...)
@@ -37,7 +37,7 @@ If any AMF instances crashes then the chunks owned by the crashed AMF instances
         * can be used by AMF pods, SMF pods

 ## Dependency
-
+
     - MongoDB should run in cluster(replicaset) Mode or sharded Mode

 ## Limitation:
@@ -45,7 +45,7 @@ If any AMF instances crashes then the chunks owned by the crashed AMF instances
       e.g. AMF can use single id for ngapid as well as tmsi

 ## Testing
-
+
     - All the DRSM clients discover other clients through pub/sub
     - Allocate resource id ( indirectly chunk). Other Pods should get notification of newly allocated chunk
     - POD down event should be detected
```

Reply:
I updated but I need a better way to try. My editor does not show any trailing spaces.

Reply:
There is no issue with spaces. Its the way you download patch file. at least when I download patch file I see space but in the code there is no space. SO I suggest you merge this and then fix those space issues.